rewrite-java-17/src/main/java/org/openrewrite/java/isolated/ReloadableJava17ParserVisitor.java

```diff
@@ -476,13 +476,31 @@ public J visitClass(ClassTree node, Space fmt) {
 
         JRightPadded<Statement> enumSet = null;
         if (!jcEnums.isEmpty()) {
-            AtomicBoolean semicolonPresent = new AtomicBoolean(false);
-
+            Tree lastConstant = jcEnums.get(jcEnums.size() - 1);
             List<JRightPadded<J.EnumValue>> enumValues = convertAll(jcEnums, commaDelim, t -> {
-                // this semicolon is required when there are non-value members, but can still
-                // be present when there are not
-                semicolonPresent.set(positionOfNext(";", '}') > 0);
-                return semicolonPresent.get() ? sourceBefore(";", '}') : EMPTY;
+                if (t != lastConstant) {
+                    return whitespace();
+                }
```

Comment:
This if-block is actually not necessary, as we are here inside the `suffix` function which is only ever applied to the last element.

Reply:
I agree but on the other hand it also guarantees that we don't accidentally do harm when it is called on an element that is not the last one